projects/storefrontlib/src/cms-components/asm/customer-selection/customer-selection.component.ts

```diff
@@ -47,23 +54,27 @@ export class CustomerSelectionComponent implements OnInit, OnDestroy {
       this.form.controls.searchTerm.valueChanges
         .pipe(debounceTime(300))
         .subscribe(value => {
-          if (!!this.selectedCustomer && value !== this.selectedCustomer.name) {
-            this.selectedCustomer = undefined;
-          }
-          if (!!this.selectedCustomer) {
-            return;
-          }
-          this.asmService.customerSearchReset();
-          if (value.trim().length >= 3) {
-            this.asmService.customerSearch({
-              query: value,
-              pageSize: this.config.asm.customeSearch.maxResults,
-            });
-          }
+          this.handleSearchTerm(value);
         })
     );
   }
 
+  private handleSearchTerm(value: string) {
+    if (!!this.selectedCustomer && value !== this.selectedCustomer.name) {
+      this.selectedCustomer = undefined;
+    }
+    if (!!this.selectedCustomer) {
+      return;
+    }
+    this.asmService.customerSearchReset();
```

Comment:
I think we need to discuss this reset for value changes because when we close the asm suggestions when we click elsewhere, we should be able to back to it when we select the input and not when the value changes.

Reply:
Ad discussed we are going to align with the accelerator version and not trigger a request on focus.